src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,567 +51,718 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+                    }
+                }  else {
+                    if (UTIL_CustomSettingsFacade.getSettings().Automatic_Household_Naming__c == TRUE) {
+                        accountIdToRename.add(oldContacts[i].AccountId);
                     }
+                }
 
+                //Old HH Accounts that don't get deleted still need to have name redone
+                if (oldContacts[i].AccountId != NULL) {
+                    accountIdToupdate.add(oldContacts[i].AccountId);
+                    contactIdAccountIdOldOppsToMove.put(c.Id, oldContacts[i].AccountId);
                 }
-                i += 1;
             }
 
-            // Must set flags after contact for loop to make sure all contacts are processed.
-            if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
-                // Set reentrant flag
-                TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Insert, true);
+            //If Ownership has changed
+            if (c.OwnerId != oldContacts[i].OwnerId) {
+                contactIdContactOwnerChange.put(c.Id, c);
             }
-            if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
-                // Set reentrant flag
-                TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Update, true);
+
+            //If Contact's last name has changed and it is using admin account model
+            if (c.LastName != oldContacts[i].LastName && c.AccountId != NULL) {
+                contactsChangedLastName.add(c);
             }
-            if (triggerAction == TDTM_Runnable.Action.AfterDelete) {
-                // Set reentrant flag
-                TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Delete, true);
+
+            //If Contact Name or Exclude_from_Household_Name__c value has changed
+            if ((c.LastName != oldContacts[i].LastName ||
+                 c.FirstName != oldContacts[i].FirstName ||
+                 c.Exclude_from_Household_Name__c != oldContacts[i].Exclude_from_Household_Name__c) &&
+                c.AccountId != NULL &&
+                checkHouseholdNaming(c.Account.RecordTypeId))
+            {
+                accountIdToRename.add(c.AccountId);
             }
         }
+    }
 
-        // now done going thru all contacts, do our DML logic based on our lists
+    /*******************************************************************************************************
+    * @description This method checks if any of the after update collections are not empty.
+    * If not empty, it calls the appropriate methods to process the list/set/map.
+    * param updateWrapper contains the collections processed from after update context.
+    * param contactsNeedAccounts a list of Contacts that needs new Accounts or an update.
+    * @return void.
```

Comment:
Same comments here re: not needing to doc the void return and preferring List notation